### PR TITLE
Projects page cleanup

### PR DESCRIPTION
**Pull request by maddiedierker**
This is a pure refactor, so there should be no user-visible changes here.

Both `projects/index.html.haml` and `projects/index.js` had a lot of artifacts from our AngularJS days, so I'm cleaning that up here to simplify the code (end goal is to add a "My Libraries" tab to the projects/ page).

**Follow-up work:** 
1. Rename `<GallerySwitcher/>` to `<ProjectsGallery/>` now that this component renders both the gallery tabs and the content -- #33121
2. Move `<PublishDialog/>` and `<DeleteProjectDialog/>` into `<PersonalProjectsTable/>`

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/sites/studio/pages/projects/index.js [hunk 1]
```diff
@@ -3,123 +3,52 @@ import React from 'react';
...
+      <div>
+        <ProjectHeader
+          canViewAdvancedTools={projectsData.canViewAdvancedTools}
+          projectCount={projectsData.projectCount}
+        />
+        <GallerySwitcher
+          limitedGallery={projectsData.limitedGallery}
+          canShare={projectsData.canShare}
+        />
+        {/* TODO: Move components below into <PersonalProjectsTable/>? */}
```
Comment (Erin007) [inline apps/src/sites/studio/pages/projects/index.js:47]: Those Dialogs are only relevant from the Personal Project Table, so sounds good to me!

**Review by Erin007 — APPROVED** [review]
Yay! Thanks for taking the time to refactor this and all of the little clean up along the way. Lovely!

File: apps/src/sites/studio/pages/projects/index.js (hunk 1 above)
Comment (maddiedierker) [reply apps/src/sites/studio/pages/projects/index.js:47]: lol i almost forgot about this! i'll make a follow-up PR to scooch these over 🎉

**Review by clareconstantine — APPROVED** [review]
I'm not very familiar with this area yet, but this makes sense to me and seems cleaner!

**Pull request merged**